test/utils/init.js

```diff
@@ -1,7 +1,26 @@
+import React from 'react';
 import enzyme from 'enzyme/build/index';
 import Adapter from 'enzyme-adapter-react-16';
 import consoleError from './consoleError';
+import { useIsSsr } from './RenderMode';
 
 consoleError();
 
+const useActualLayoutEffect = React.useLayoutEffect;
+
+/* eslint-disable react-hooks/rules-of-hooks */
+/**
+ * during tests we are in a jsdom environment but will also call ReactDOM.renderToString
+ * which triggers useLayoutEffect server warnings. On an actual server there's
+ * no jsdom
+ */
+// eslint-disable-next-line camelcase
+React.useLayoutEffect = function unstable_useIsomorphicLayoutEffect(fn, deps) {
```

Comment:
Tried creating a separate ssr test suite but this was too much work for a couple of tests. While this isn't pretty it works.

Reply:
I have used `process.env.NODE_ENV !== 'test'` in NoSsr.js. I believe we should consolidate around one solution.

Reply:
But this will use `useLayoutEffect` in properly setup ssr test setups i.e. for test setups without JSDOm but with `NODE_ENV=test` this will trigger useLayoutEffect warnings